### ISSUE TITLE
gnome-shell crash when locking screen

### ISSUE THREAD

**Issue by fxha** [ewlsh/dynamic-panel-transparency#108]
First of all thank you for this extension. I discovered a `gnome-shell` crash when locking screen that result from this extension. Please see [here](https://gitlab.gnome.org/GNOME/gnome-shell/issues/1265#note_675710) for more details. Everything works fine when `Dynamic Panel Transparency` is disabled, so I opened this issue but I guess there is also a bug in GNOME.

GNOME Shell version: at least 3.32.2, ..., 3.34.2

**How to reproduce:**

1. Install `Dynamic Panel Transparency` extension.
2. Launch a not GTK application like Firefox or Chromium but not gedit or Nautilus.
3. Maximize the application and close the window or unmaximize it and  press `Ctrl+L` immediately.
4. `gnome-shell` is crashed.

**Comment by ZahnPastor** [comment]
can confirm this bug too, tried to lock the screen, didnt happen, gnome became unresponsive, for it only to crash and restart a few seconds later

using lockscreen works perfectly now, but whenever i try to enable the extension now i get an error

**Comment by rockon999** [comment]
I am willing to bet this is related to DPT's attempt to "blend-in" to the user's theme.

In future versions of the extension, this feature will likely be removed as it is challenging to support and in the case user's want to use custom themes they will be able to set the panel colors manually.

**Comment by AlexKF98** [comment]
For me it crashes about 50% of the time when locking screen, including when I close my laptop's lid. This is actually a security problem, because the laptop doesn't really lock and when the lid is opened again it bypasses the lockscreen and shows gnome-shell crashed.

**Comment by yumasheta** [comment]
+1
running gnome-shell 3.36.5-1 on archlinux.
gnome-shell segfaults:
`error 5 in libmutter-6.so.0.0.0[7f35b7d3d000+14b000]`
and the stack trace starts with:
```
.local/share/gnome-shell/extensions/dynamic-panel-transparency@rockon999.github.io/extension.js:162 (159edd2a43d0 @ 27)
.local/share/gnome-shell/extensions/dynamic-panel-transparency@rockon999.github.io/events.js:139 (159edd2aa808 @ 570)
```
I am with @AlexKF98 on the security concerns. If you want to suspend the system, and the shell crashes, later when resuming, the session will not be locked.
This propably should also be reported to gnome-shell directly.

**Comment by ewlsh** [comment]
This should theoretically be fixed in #126.
